### DISCUSSION TITLE
Release v1.9.0

### DISCUSSION THREAD

**Discussion by dewenni** [dewenni/ESP32-Jarolift-Controller#1] [Announcements]
# 1.0.0

## what's new

this is the initial release of this new project 🎉🎉🎉

If you like this project, feel free to push the **[Star ⭐️]** button and click **[Watch 👁]** to stay updated.

If there is any bug, please use the Issue function and let me know.

If you have any further questions, suggestions or requests, please use the discussion function.

**Comment by dewenni** [comment]
# v1.4.0

## what's new

### 1. direct GitHub Update 🎉🎉🎉

This update introduces an exciting new feature:  
You can now update to the latest version directly from the WebUI – no need to download the .bin file manually anymore!

Simply click on the version info in the bottom left corner, and a dialog will open. If a new version is available, you can initiate the update right there. It will automatically download and install the latest release from GitHub.

![](https://raw.githubusercontent.com/dewenni/ESP32-Jarolift-Controller/main/Doc/github_ota.gif)

### 2. new option to enable/disable WIFI

there is a new option to enable/disable WiFi. This could be useful if you use the Ethernet connection and you do not want to use the WiFi in parallel.
The depending switch to enable/disable is also in Settings > WiFi

> [!CAUTION]
> Unfortunately after the update the "enable" will be false and therefore the Controller will be activate the "Setup-Mode" after restart, because WiFi and in most cases Ethernet is now disabled. Then you have to connect to the Accesspoint, go to Settings and activate WiFi. After Reboot everything should be fine again. Sorry, but that's something I can't avoid in this case.

> [!TIP]
> Maybe it is necessary to clean your browser cache after the update, to be sure that everything works well!

## changelog

- add new feature to update directly from GitHub
- add option to enable/disable WIFI in case ETH is used (WiFi will be activated automatically in Setup-Mode) #17
- improve the size of controls at mobile layout

**Reply by flushbug** [reply]
works !

**Reply by dewenni** [reply]
the new update option is quite cool in my point of view.
You can use it with the next release.

**Comment by dewenni** [comment]
# v1.5.0

## what's new

This update brings a cool new feature 🎉🎉🎉

### 1. Timer Control

The timer function enables the automatic control of individual roller shutters or a selection of several roller shutters as a group.
A fixed time can be specified as a trigger, or sunrise or sunset with an optional time offset.
This controller can therefore also work well as a stand-alone solution without the need for further automation

![webUI_timer](https://raw.githubusercontent.com/dewenni/ESP32-Jarolift-Controller/main/Doc/webUI_timer.png)

The selection of roller shutters is supported by an additional dialogue. All configured and activated roller shutters are displayed there. These can be selected there and the bitmask is then automatically created from them.

<p align="left">
    <img src="https://raw.githubusercontent.com/dewenni/ESP32-Jarolift-Controller/main/Doc/webUI_bitmask_wiz.png" alt="Description" width="400">
</p>


> [!TIP]
> Maybe it is necessary to clean your browser cache after the update, to be sure that everything works well!

---

### encrypted Passwords

Passwords are now better protected and stored in encrypted form in the config.json
When updating, the existing passwords are automatically encrypted and saved again.

> [!CAUTION]
> As the passwords are stored in encrypted form after this update, it is no longer possible to switch to an older version without re-entering the passwords after the downgrade

---

> [!TIP]
> Remember that if you are already at v1.4.0, you can also use the possibility to update to this version right out of the WebUI of your controller if you click on the version information at the bottom left of the WebUI. [See also here](https://github.com/dewenni/ESP32-Jarolift-Controller?tab=readme-ov-file#github-ota-update)



---

## changelog

- [UPDATE] Arduino core 3.1.1 based on IDF 5.3.2.241224
- [UPDATE] mathieucarbou/AsyncTCP @ 3.3.2
- [UPDATE] mathieucarbou/ESPAsyncWebServer @ 3.6.0
- [FEATURE] new feature to define 6 individual timer to automatically move the shutters #4
- [FEATURE] new wizard for defining bit masks for timers and groups
- [IMPROVE] uploaded config files via "config-upload" are now automatically renamed to config.json, regardless of the original filename.
- [IMPROVE] Improved behavior when the restart button is pressed immediately after a change in the settings.
- [IMPROVE] Passwords are better protected and are stored in encrypted form

**Comment by dewenni** [comment]
# v1.6.0

## what's new

this is a minor update and bugfix version

### WebUI-Demo

For a first impression of the functions and the WebUI, a limited demo is also available.  
This can be accessed via the following link: [WebUI-DEMO](https://dewenni.github.io/ESP32-Jarolift-Controller/)

### Changed condition for Setup-Mode

Setup mode is now activated when the ESP is restarted **5** times.
A maximum of 5 seconds may elapse after each restart.

Example: restart 1/5 - wait 2s - restart 2/5 - wait 2s - restart 3/5 - wait 2s - restart 4/5 - wait 2s - restart /5/5 => Setup-Mode


## changelog

- [UPDATE]  ESP32Async/AsyncTCP @ 3.3.5
- [UPDATE]  ESP32Async/ESPAsyncWebServer @ 3.7.0
- [IMPROVE] Added validation for IP input fields in the setup area to avoid wrong syntax #27
- [IMPROVE] Conversion of the storage of the device counter from EEPROM to NVS
- [IMPROVE] remove callback function for internal messages in jarolift-controller library
- [IMPROVE] change "Double-Reset-Detection" to "Multi-Reset-Detection" now it needs 5 restarts within 5s timeout to enter Setup-Mode #26
- [NEW] Add a [WebUI-DEMO](https://dewenni.github.io/ESP32-Jarolift-Controller/) under github-pages
- [NEW] Added some description for possible teach-in procedures.

**Comment by dewenni** [comment]
# v1.6.1

## what's new

this is a minor bugfix version


## changelog

- [BUGFIX] URL entries in the settings were not saved #29

**Comment by dewenni** [comment]
# v1.6.2

## what's new

this is a minor bugfix version


## changelog

- [BUGFIX] Login problems due to faulty cookie handling fixed and improved #30

**Comment by dewenni** [comment]
# v1.6.3

## what's new

this is a minor bugfix version


## changelog

- [IMPROVE] improved Login Form Feedback for Incorrect Credentials
- [IMPROVE] add build script to automatically update version in github-pages
- [BUGFIX] fix missing log entries from jarolift-controller library #29 #33
- [BUGFIX] device counter was not saved after editing in WebUI settings #33

**Comment by dewenni** [comment]
## what's new

### EspWebUI-Library

with this update, the basic WebUI functions were outsourced to a separate [EspWebUI](https://github.com/dewenni/EspWebUI) library.
This should have no effect on the user experience, but it improves the handling of similar projects like [ESP_Buderus_KM271](https://github.com/dewenni/ESP_Buderus_KM271)

### support for received remote signals

As requested by some users, there is now also support for original remote controls. The signals from the remote controls are now captured, processed and also made available via MQTT. Further details can be found in the updated and extended readme.

### support for different ESP32 Chips

There is now also support for other ESP32 chips. This allows you a greater variance in the selection of your hardware.
For the newly supported chips, a firmware file for OTA updates and one for initial flashing will be offered in future. The direct GitHub update via the WebUI should also work for the new chips.

Supported Chips:

- ESP32
- ESP32-S2 (new)
- ESP32-S3 (new)
- ESP32-C3 (new)

### some minor WebUI improvements

- When a shutter command is executed in the WebUI, a message confirming the command is displayed for 2 seconds at the footer.
- Add additional ESP-Chip infos in the system section of the WebUI
- Add add additional error messages at the footer if Jarolift Keys or serial number not set

## changelog

- [UPDATE]  ESP32Async/AsyncTCP @ 3.3.6
- [UPDATE]  ESP32Async/ESPAsyncWebServer @ 3.7.2
- [UPDATE]  bblanchon/ArduinoJson @ 7.3.1
- [CHANGE] Basic WebUI functions were outsourced to a separate [EspWebUI](https://github.com/dewenni/EspWebUI) library
- [FIX] update WiFi and Ethernet Status also if mqtt is not connected
- [FIX] the entry of Jarolift Keys or the serial number was not accepted if you clicked on LOCK directly after the entry
- [FIX] skip websocket connection if WebUI is located on localhost or github.io
- [FEATURE] add additional error messages if Jarolift Keys or serial number not set
- [FEATURE] When a shutter command is executed in the WebUI, a message confirming the command is displayed for 2 seconds at the footer.
- [FEATURE] add support for received remote signals
- [FEATURE] add support for multiple ESP32 Chips
- [FEATURE] add additional ESP-Chip infos in the system section of the WebUI

**Comment by dewenni** [comment]
# v1.9.0

## what's new

### Service Commands

there is a new WebUI Page with service commands like:

- set shade
- set/delete upper end point
- set/delete lower end point
- unlearn (located in settings)

### remote signals (update)

The MQTT message for remote signals has been updated. The information about which shutter is controlled can be seen in the two variables `chBin` and `chDec`. `chBin` shows the used shutter as a 16-bit binary value, while `chDec` shows the same information as a decimal value for easier automation.

```
topic:      ../status/remote/<serial-number>
payload:    {
              "name":   "<alias-name>", 
              "cmd":    "<UP, DOWN, STOP, SHADE>",
              "chBin":  "<channel-binary>",
              "chDec":  "<channel-decimal>"
            }
```

## changelog

- [FIX] bugfix github ota asset check #40, #45
- [FEATURE] service page with new service commands
- [FEATURE] new "unlearn" command in shutter settings
- [CHANGE] "set shade" was moved from settings to service page
- [CHANGE] MQTT messages for remote signals have been updated
- [CHANGE] internal redesign in JaroliftController-Lib
- [UPDATE] dewenni/EspWebUI @ 0.0.4

**Comment by RobbiTobbi1955** [comment]
Hi,
1st of all: you did a really great job, thank you very much!

I build the hardware as described, flashed it with 1.9.0, configured HA and everything worked from the beginning, no problems whatsoever.
I now disconned the jalousiescout bridge and will sell it. This solution is much, much better!

But after some days I missed an indication for "communicating" and wanted this little blue LED blink when a mqtt message is sent or received. So i took your source code and tried to compile. 

There were two errors:
a) In platform.ini you have the versions for the web UI and the current version is different from it. 
Solution: I removed the version text completely from platform.ini.
b) In lang.js the texts for "log_mode_1".."log_mode_4" is missing, but used by the html, so there is a second error.
Added these texts and could compile.

The resulting bin is OK, so the compilation is indeed successful then.

But neverthelesse, be it the original or the "private" flash: 

I built the same hardware 2 times; on is actually running and does its job; the second one is for redundancy only. 

But when testing I realized a problem: 

(->description)
1st to know: one has the adress x.x.x.10, the other has the adress x.x.x.11, first uses "jarolift" as mqtt-topic plus HA active. The second uses "esp-jarolift" as topic and no homeasisstant (with/without makes no difference for the problem).

Problem: I can NOT connect with both devices at the same time to the mosquito mqtt server. 
To connect the second one I must deactivate the first mqtt-connection and vice versa. 

Both use the same "user" and "password", as many of my devices do, so that I can connect them to the mosquito. But even when I add another user "jarolift" to the mosquito and give these data to the second device the problem persists.

(->solution)
I found out that the problem comes from the host name ("ClientID"). 
Changing the wifi host name to something else (ESP-Jarolift to ESP-jarolift2) worked immediately.
adding these lines to mqttSetup() helped to find out:

  ESP_LOGD(TAG, "MQTT   server is %s:%d", config.mqtt.server, config.mqtt.port);
  ESP_LOGD(TAG, "ClientID name is %s", config.wifi.hostname);
  ESP_LOGD(TAG, "Username      is %s", config.mqtt.user);
  ESP_LOGI(TAG, "MQTT setup done!");

But beyond from this, I want to thank you again!

kind regards
und vielen Dank
Robert Hörner, Philippsburg

**Reply by dewenni** [reply]
Hi Robert,

you are right. The mqtt clientID is based on the WiFi-Hostname.
https://github.com/dewenni/ESP32-Jarolift-Controller/blob/067f0ad3dc0de47eac7edb03d88ca2b9c72c22cb/src/mqtt.cpp#L190

Honestly this is maybe not the best way to do.
An alternative would be to derive this from the chip ID or Mac ID in order to obtain a device-dependent and unique client ID.

On the other hand, it is not ideal to use the same host name for multiple devices.
